include/eve/detail/dispatcher.hpp

```diff
@@ -0,0 +1,51 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+#include <utility>
+
+namespace eve::detail
+{
+  template<typename Tag, typename Dummy = void> struct  callable_object_bis;
+
+  template <typename Tag, typename... Args>
+  concept tag_dispatchabe = requires(Tag tag, Args&&... args)
+  {
+    { tagged_dispatch(tag, std::forward<Args>(args)...) };
+  };
```

Comment:
Hmm. I'm actaully not sure why we need `do_tagged_dispatch`
Wouldn't concept guarntee adl lookup?